ClientMapStoreTest.mapSize_After_MapStore_OperationQueue_OverFlow_Test

```
junit.framework.AssertionFailedError: expected:<50000> but was:<50001>
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.Assert.failNotEquals(Assert.java:329)
	at junit.framework.Assert.assertEquals(Assert.java:78)
	at junit.framework.Assert.assertEquals(Assert.java:234)
	at junit.framework.Assert.assertEquals(Assert.java:241)
	at com.hazelcast.client.map.ClientMapStoreTest.mapSize_After_MapStore_OperationQueue_OverFlow_Test(ClientMapStoreTest.java:142)
```
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.maintenance-nightly/com.hazelcast$hazelcast-client/508/testReport/junit/com.hazelcast.client.map/ClientMapStoreTest/mapSize_After_MapStore_OperationQueue_OverFlow_Test/

**sonar**
```
java.lang.AssertionError: expected:<50000> but was:<50001>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.client.map.ClientMapStoreTest.mapSize_After_MapStore_OperationQueue_OverFlow_Test(ClientMapStoreTest.java:142)
```

https://hazelcast-l337.ci.cloudbees.com/view/Sonar/job/Hazelcast-3.x-sonar/com.hazelcast$hazelcast-client-legacy/898/testReport/junit/com.hazelcast.client.map/ClientMapStoreTest/mapSize_After_MapStore_OperationQueue_OverFlow_Test/

--- a/hazelcast-client-legacy/src/test/java/com/hazelcast/client/map/ClientMapStoreTest.java
+++ b/hazelcast-client-legacy/src/test/java/com/hazelcast/client/map/ClientMapStoreTest.java
@@ -23,12 +23,15 @@ import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
 import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
 
 import static org.junit.Assert.assertEquals;
 
@@ -116,8 +119,8 @@ public class ClientMapStoreTest extends HazelcastTestSupport {
         MapConfig mapConfig = new MapConfig();
         MapStoreConfig mapStoreConfig = new MapStoreConfig();
 
-        final MapStoreBackup store = new MapStoreBackup();
-        final int delaySeconds = 4;
+        MapStoreBackup store = new MapStoreBackup();
+        int delaySeconds = Integer.MAX_VALUE;
         mapStoreConfig.setEnabled(true);
         mapStoreConfig.setImplementation(store);
         mapStoreConfig.setWriteDelaySeconds(delaySeconds);
@@ -130,16 +133,27 @@ public class ClientMapStoreTest extends HazelcastTestSupport {
 
         HazelcastInstance server = hazelcastFactory.newHazelcastInstance(config);
         HazelcastInstance client = hazelcastFactory.newHazelcastClient();
-        final IMap map = client.getMap(MAP_NAME);
+        IMap map = client.getMap(MAP_NAME);
 
-
-        final int max = getMaxCapacity(server) + 1;
-        for (int i = 0; i < max; i++) {
-            map.putAsync(i, i);
+        int maxCapacity = getMaxCapacity(server);
+        List<Future> futures = new ArrayList<Future>(maxCapacity * 2);
+        for (int i = 0; i < maxCapacity * 2; i++) {
+            Future future = map.putAsync(i, i);
+            futures.add(future);
         }
 
-        Thread.sleep(1000 * (delaySeconds + 1));
-        assertEquals(max - 1, map.size());
+        int success = 0;
+        for (Future future : futures) {
+            try {
+                future.get();
+                success++;
+            } catch (ReachedMaxSizeException e) {
+                //ignore
+            }
+        }
+
+        assertEquals(success, maxCapacity);
+        assertEquals(map.size(), maxCapacity);
     }
 
     @Test(expected = ReachedMaxSizeException.class)

--- a/hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapStoreTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapStoreTest.java
@@ -23,12 +23,15 @@ import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
 import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
 
 import static org.junit.Assert.assertEquals;
 
@@ -116,8 +119,8 @@ public class ClientMapStoreTest extends HazelcastTestSupport {
         MapConfig mapConfig = new MapConfig();
         MapStoreConfig mapStoreConfig = new MapStoreConfig();
 
-        final MapStoreBackup store = new MapStoreBackup();
-        final int delaySeconds = 4;
+        MapStoreBackup store = new MapStoreBackup();
+        int delaySeconds = Integer.MAX_VALUE;
         mapStoreConfig.setEnabled(true);
         mapStoreConfig.setImplementation(store);
         mapStoreConfig.setWriteDelaySeconds(delaySeconds);
@@ -130,16 +133,27 @@ public class ClientMapStoreTest extends HazelcastTestSupport {
 
         HazelcastInstance server = hazelcastFactory.newHazelcastInstance(config);
         HazelcastInstance client = hazelcastFactory.newHazelcastClient();
-        final IMap map = client.getMap(MAP_NAME);
+        IMap map = client.getMap(MAP_NAME);
 
-
-        final int max = getMaxCapacity(server) + 1;
-        for (int i = 0; i < max; i++) {
-            map.putAsync(i, i);
+        int maxCapacity = getMaxCapacity(server);
+        List<Future> futures = new ArrayList<Future>(maxCapacity * 2);
+        for (int i = 0; i < maxCapacity * 2; i++) {
+            Future future = map.putAsync(i, i);
+            futures.add(future);
         }
 
-        Thread.sleep(1000 * (delaySeconds + 1));
-        assertEquals(max - 1, map.size());
+        int success = 0;
+        for (Future future : futures) {
+            try {
+                future.get();
+                success++;
+            } catch (ReachedMaxSizeException e) {
+                //ignore
+            }
+        }
+
+        assertEquals(success, maxCapacity);
+        assertEquals(map.size(), maxCapacity);
     }
 
     @Test(expected = ReachedMaxSizeException.class)

